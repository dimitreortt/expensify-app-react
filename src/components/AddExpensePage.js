import React from 'react'
import ExpenseForm from './ExpenseForm'
import { connect } from 'react-redux'
import { addExpense } from '../actions/expenses'
import { setTextFilter } from '../actions/filters'

const AddExpensePage = (props) => (
  <div>
    <h1>Add Expense</h1>
    <ExpenseForm
      onSubmit={(expense) => {
        console.log(expense, 'oaushdkaushd  ')
        console.log(addExpense(expense))
        props.dispatch(addExpense(expense))
        console.log(addExpense(expense))
        props.history.push('/help')
      }}
    />
  </div>
)

export default connect()(AddExpensePage)