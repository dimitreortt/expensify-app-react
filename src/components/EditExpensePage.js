import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import React, { Component } from 'react'

export class EditExpensePage extends Component {
  onRemoveButtonClick = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id })
    this.props.history.push('/')
  }
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  }
  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button onClick={this.onRemoveButtonClick}>Remove</button>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  }
}

const mapDispatchToProps = (dispatch) => ({
  startRemoveExpense: (id) =>
    dispatch(startRemoveExpense(id))
  ,
  startEditExpense: (id, expense) =>
    dispatch(startEditExpense(id, expense))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)
