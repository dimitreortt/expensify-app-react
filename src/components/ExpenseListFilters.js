import { connect } from 'react-redux'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'
import React, { Component } from 'react'
import { DateRangePicker } from 'react-dates'

export class ExpenseListFilters extends Component {
  state = {
    calendarFocused: null
  }
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate)
    this.props.setEndDate(endDate)
  }
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }))
  }
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value)
  }
  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate()
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount()
    }
  }
  render() {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              placeholder="Search expenses"
              className="text-input"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="input-group__item">
            <select
              className="select"
              value={this.props.filters.sortBy}
              onChange={this.onSortChange}
            >
              <option value="date">Date</option>
              <option value="amount">Amount</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDate={this.props.filters.startDate}
              endDate={this.props.filters.endDate}
              onDatesChange={this.onDatesChange}
              focusedInput={this.state.calendarFocused}
              onFocusChange={this.onFocusChange}
              numberOfMonths={1}
              showClearDates={true}
              isOutsideRange={() => false}
            />
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (value) => dispatch(setTextFilter(value)),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  sortByAmount: () => dispatch(sortByAmount()),
  sortByDate: () => dispatch(sortByDate())
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)
