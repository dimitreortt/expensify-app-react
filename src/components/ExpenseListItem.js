import React from 'react'
import { Link, NavLink } from 'react-router-dom'
import moment from 'moment'

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
  <div>
    <NavLink to={'edit/' + id}>
      <h3>{description}</h3>
    </NavLink>
    <p>
      {amount}{' - '}
      {moment(createdAt).format('MMMM Do, YYYY')}
    </p>
  </div>
)

export default ExpenseListItem
