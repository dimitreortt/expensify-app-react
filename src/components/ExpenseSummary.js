import selectExpensesTotal from "../selectors/expenses_total"
import selectExpenses from "../selectors/expenses"
import React from 'react'
import { connect } from "react-redux"

export const ExpenseSummary = ({ expensesCount, expensesTotal }) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'

  return (
    <div>
      <h3>Viewing {expensesCount} {expenseWord} totalling {expensesTotal}</h3>
    </div>
  )
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)

  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpenseSummary)