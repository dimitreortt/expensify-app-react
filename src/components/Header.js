import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
  <header>
    <h3>Expensify</h3>
    <NavLink activeClassName="is-active" to="/" exact>Home</NavLink>
    <NavLink activeClassName="is-active" to="/create">Add Expense</NavLink>
    <NavLink activeClassName="is-active" to="/edit">Edit Expense</NavLink>
    <NavLink activeClassName="is-active" to="/help">Help</NavLink>
  </header>
)

export default Header