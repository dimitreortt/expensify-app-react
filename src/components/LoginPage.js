import React from 'react'
import { startLogin } from '../actions/auth'
import { connect } from 'react-redux'

export const LoginPage = ({ startLogin }) => (
  <div>
    <h1>Please Log In</h1>
    <button onClick={startLogin}>Login</button>
  </div>
)

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)