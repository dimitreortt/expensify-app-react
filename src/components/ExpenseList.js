import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseList = (props) => (
  <div>
    <h2>Expense List</h2>
    <ExpenseListFilters />
    {props.expenses.map((expense) => {
      console.log(expense)
      console.log(expense)
      return <ExpenseListItem key={expense.id} {...expense} />
    })}
  </div>
)

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  }
}

export default connect(mapStateToProps)(ExpenseList)
