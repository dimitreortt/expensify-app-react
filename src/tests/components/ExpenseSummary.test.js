import { shallow } from "enzyme"
import { ExpenseSummary } from '../../components/ExpenseSummary'
import React from 'react'

test('should render ExpenseSummary correctly with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={124} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary correctly with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expensesCount={3} expensesTotal={25245} />)
  expect(wrapper).toMatchSnapshot()
})