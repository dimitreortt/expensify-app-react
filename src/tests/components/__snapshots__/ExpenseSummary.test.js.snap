// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with 1 expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    124
  </h3>
</div>
`;

exports[`should render ExpenseSummary correctly with 1 expense 2`] = `
<div>
  <h3>
    Viewing 
    3
     
    expenses
     totalling 
    25245
  </h3>
</div>
`;
